### PR TITLE
vala: patch to circumvent apparent pkgconf bug

### PR DESCRIPTION
**Pull request by flyn-org**
There seems to be a bug in pkgconf 1.9 that broke building the OpenWrt Vala package. This adds a patch to circumvent the bug. For the related discussion, see:

https://github.com/openwrt/packages/issues/21359

The pkgconf bug appears similar to the one discussed here:

https://github.com/pkgconf/pkgconf/issues/268

Maintainer: me
Compile tested: x86_64

### CODE REVIEW

File: lang/vala/patches/001-circumvent-glib-pkgconf-bug [hunk 1]
```diff
@@ -0,0 +1,75 @@
+This patch circumvents an apparent bug in pkgconf 1.9 that prevents
+building the OpenWrt Vala package. For a discussion of this, see:
+
+https://github.com/openwrt/packages/issues/21359
+
+The pkgconf bug seems similar to the one discussed here:
+
+https://github.com/pkgconf/pkgconf/issues/268
+
+diff -u --recursive vala-0.56.3-vanilla/configure vala-0.56.3/configure
```
Comment (1715173329) [inline lang/vala/patches/001-circumvent-glib-pkgconf-bug:10]: Please refresh this patch and add proper suffix to filename.

Comment (1715173329) [reply lang/vala/patches/001-circumvent-glib-pkgconf-bug]: To refresh it:
`make package/vala/host/{clean,refresh} V=s`

**Comment by flyn-org** [comment]
@ 1715173329, i refreshed the patch. I was not familiar with the target for host packages until you provided the command. Thank you!

**Comment by feckert** [comment]
@flyn-org If you want to be the maintainer of the vala package. Then please add yourself as a maintainer in a separate commit.

File: lang/vala/Makefile [hunk 2]
```diff
@@ -11,6 +11,8 @@ PKG_NAME:=vala
 PKG_VERSION:=0.56.3
 PKG_RELEASE:=1
 
+PKG_MAINTAINER:=W. Michael Petullo <mike@flyn.org>
```
Comment (1715173329) [inline lang/vala/Makefile:14]: consider updating line 20 instead :-)

**Review by 1715173329 — APPROVED** [review]

**Comment by 1715173329** [comment]
if possible please create a pr for the 23.05 branch too, thanks!

**Comment by flyn-org** [comment]
See https://github.com/openwrt/packages/pull/21473.

**Pull request merged**